docs/core/diagnostics/logging-tracing.md

```diff
@@ -0,0 +1,94 @@
+
+The choice of which API to use is up to you. The key differences:
+- <xref:System.Console?displayProperty=nameWithType>
+  - Always compiled in and always writes to the console.
+  - Useful for information that your customer may need to see in release.
+  - Because it's the simplest approach, it's often used for ad-hoc temporary debugging. This debug code is often never checked in to source control.
+- <xref:System.Diagnostics.Trace?displayProperty=nameWithType>
+  - Only enabled when `TRACE` is defined.
+  - Writes to attached <xref:System.Diagnostics.TraceListener>s. Usually a debugger.
+  - It seems best to use this API when creating release-oriented logs.
+- <xref:System.Diagnostics.Debug?displayProperty=nameWithType>
+  - Only enabled when `DEBUG` is defined.
+  - Writes to attached <xref:System.Diagnostics.TraceListener>s. Usually a debugger.
+  - It seems best to use this API when creating debug-oriented logs.
+
+### Logging Related References
+
+- [How to: Compile Conditionally with Trace and Debug](../../framework/debug-trace-profile/how-to-compile-conditionally-with-trace-and-debug)
+
+- [How to: Add Trace Statements to Application Code](../../framework/debug-trace-profile/how-to-add-trace-statements-to-application-code)
+
+- [ASP.NET Logging](/aspnet/core/fundamentals/logging)
+provides an overview of the logging techniques it supports.
+
+- [C# String Interpolation](../../csharp/language-reference/tokens/interpolated)
+ can simplify writing logging code.
+
+- The <xref:System.Exception.Message?displayProperty=nameWithType> is useful for logging exceptions.
+
+- The <xref:System.Diagnostics.StackTrace?displayProperty=nameWithType>
+can be useful to provide debugger like logging info.
+
+- <xref:System.Diagnostics.TraceListener> provides an overview of existing listeners and creating a custom listener.
+
+## Tracing APIs
+
+- <xref:System.Diagnostics.Tracing.EventSource?displayProperty=nameWithType>
+  - EventSource is the primary root .NET Core tracing API
+  - Available in all .NET Standard versions
+  - Emits to .NET Core's event pipe on all platforms
+  - Also emits to ETW on Windows.
+  - Only allows tracing serializable objects
+
+- <xref:System.Diagnostics.DiagnosticSource?displayProperty=nameWithType>
+  - Only available in .NET Core
+  - Allows tracing non-serializable objects
+  - Includes a bridge to allow select fields of logged objects to be written to an <xref:System.Diagnostics.Tracing.EventSource>.
+
+- <xref:System.Diagnostics.TraceSource?displayProperty=nameWithType>
+  - This API was introduced in .NET Framework 2.0
+  - Supported in .NET Standard 2.0
+  - Prefer <xref:System.Diagnostics.Tracing.EventSource>
+
+- <xref:System.Diagnostics.EventLog?displayProperty=nameWithType>
+  - Windows only
+  - Writes messages to Windows event log
+  - System administrators expect fatal application error messages to appear in the Windows event log.
+
+## Performance Considerations
```

Comment:
I'm wondering if we need this section or if we should put it somewhere else? Most of the page described options available for logging and then here we are switching to some best practices for authoring logging.

Reply:
```suggestion
## Performance considerations
```